### DISCUSSION TITLE
Is CodeQL necessary to use custom piece repositories?

### DISCUSSION THREAD

**Discussion by tdari** [Tauffer-Consulting/domino#221] [Q&A] [answered]
CodeQL is available for free accounts only in public repositories. Can I use private piece repositories if I remove all actions and release manually?

**Comment by vinicvaz** [comment]
Hey @tdari .
No, codeql is not necessary to release the pieces. 
Yes, you can remove all the actions and release manually if you want.
Is the domino release action also strict to public repos?

**Reply by tdari** [reply]
When I delete CodeQL no workflow runs. I'm quite new to Github Packages but it seems like i need to publish a package for docker to build? Because when i manually publish a release even though i changed the metadata in the template it still builds from the template and it cause this error:

```
Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='100m', input_type=str]
For further information visit https://errors.pydantic.dev/2.4/v/int_parsing
requests.memory
Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='128Mi', input_type=str]
For further information visit https://errors.pydantic.dev/2.4/v/int_parsing
limits.cpu
Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='500m', input_type=str]
 For further information visit https://errors.pydantic.dev/2.4/v/int_parsing
```

![image](https://github.com/Tauffer-Consulting/domino/assets/50070971/7e7b10b0-0292-48eb-8073-416d20bf8ed7)

**Comment by vinicvaz** [comment, marked as answer]
Hello @tdari 
Probably the workflow is not running because it is only triggered when you update the `config.toml` file in the pieces repository root, it is the way we are ensuring that no actions are being triggered when no version update was done.

Please check if you filled the `config.toml` file with the right information, example:
```

REGISTRY_NAME = "tauffer-consulting" # Here you should put your registry name, if you are using your personal account, it should be your github username

REPOSITORY_NAME = "default_domino_pieces" # you can use the same name as it is in your repository url
REPOSITORY_LABEL = "Default Pieces Repository" # Just a label, it will be shown in the UI

VERSION = "0.1.0" # The version for the release of the repository
```
When you commit this file your action will be triggered, even without codeql.

---

About the second error (the validation one), was our mistake on having our template repository outdated. 
The values for `container_resources` in the `metadata.json` files should now be **integer**, not **strings**.
I just updated the template repository, you can use it or you can replace manually your values from `m̀etadata.json` like

```
    "container_resources": {
        "requests": {
            "cpu": 100, # from "100m"
            "memory": 128 # from "100Mi"
        },
        "limits": {
            "cpu": 500, # from "500m"
            "memory": 512 # from "500Mi"
        }
    },
```
Notice that cpu and memory values are **integers** not strings.

---


### **IMPORTANT**
Please, go to your repository Settings > Actions > General > Workflow permissions
And give **Read and Write permission**
This will allow actions to git commit and push to your branch.


Thanks

**Reply by tdari** [reply]
Finally I managed to publish :) I appreciate your detailed answer I wasn't aware of workflow is triggered by the config file.

**Reply by vinicvaz** [reply]
Great! I'll add this info to the docs, I don't think we have it there.
Thanks
